modules/core/src/main/java/org/apache/ignite/internal/client/thin/ReliableChannel.java

```diff
@@ -817,7 +819,11 @@ private <T> T applyOnDefaultChannel(
         ClientOperation op,
         @Nullable List<ClientConnectionException> failures
     ) {
-        while (attemptsLimit > (failures == null ? 0 : failures.size())) {
+        // +1 is required for the correct channel search if ReliableChannel#applyOnDefaultChannel#idx selects
```

Comment:
An additional attempt is needed because N+1 channels might be used for sending a message - first a random one, then each one from #channels in sequence.

Reply:
Wouldn't it be better to explain why this result of a random choice influences?